dashboard/test/models/pd/teacher_application_test.rb

```diff
@@ -53,6 +54,14 @@ class Pd::TeacherApplicationTest < ActiveSupport::TestCase
       create :pd_teacher_application, secondary_email: 'invalid@ example.net'
     end
     assert_equal 'Validation failed: Secondary email does not appear to be a valid e-mail address', e.message
+
+    e = assert_raises ActiveRecord::RecordInvalid do
+      application = create :pd_teacher_application
+      hash = application.application_hash
+      hash['principalEmail'] = 'invalid@ example.net'
+      application.update!(application: hash.to_json)
+    end
```

Comment:
This should only fail in the `update!` (line 62). Lines 59-61 can go outside the `assert_raises` block.